### PR TITLE
Add StandardJS code style linter

### PR DESCRIPTION
**Pull request by zuzak**
This commit fixes a few interesting code style decisions, and adds a linter to prevent it happening again.

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
+/* eslint-env mocha */
 
 const domains = require('..')
 const assert = require('assert') // core
 const isValidDomain = require('is-valid-domain')
 
 describe('module exports', function () {
   it('should export an array', function () {
-    assert.ok(Array.isArray(domains));
+    assert.ok(Array.isArray(domains))
```
Comment (sulthan-ahmed) [inline test/test.js]: why no `;`?

File: test/test.js [hunk 2]
```diff
@@ -1,19 +1,18 @@
...
   })
 })
 
-
 describe('individual emails', function () {
-  for (i in domains) {
-    let domain = domains[i]
+  for (const i in domains) {
+    const domain = domains[i]
```
Comment (sulthan-ahmed) [inline test/test.js]: how is this `const`? the variable changes

**Comment by sulthan-ahmed** [comment]
good to add a linter 👍 just wondering how this is better than `es-lint`?

File: test/test.js (hunk 2 above)
Comment (sulthan-ahmed) [reply test/test.js]: ok I learnt something so this is actually the right thing to do 👍  https://medium.com/@mautayro/es6-variable-declaration-for-loops-why-const-works-in-a-for-in-loop-but-not-in-a-normal-a200cc5467c2

File: test/test.js (hunk 1 above)
Comment (zuzak) [reply test/test.js]: Consistency. Either every line needs a semi-colon, or none.
(StandardJS -- a fairly arbitrary code linter -- has opinionated defaults so people don't waste time arguing about semicolons or not.)

File: test/test.js (hunk 2 above)
Comment (zuzak) [reply test/test.js]: `const` is (counterintuitively) valid here -- `const` is block scoped. It doesn't exist outside of its `{}`, so on each iteration it gets redeclared. As one should always use `const` over `let` over `var` when possible,. `const` wins out.

Someone wrote a [blog post about it](https://medium.com/@mautayro/es6-variable-declaration-for-loops-why-const-works-in-a-for-in-loop-but-not-in-a-normal-a200cc5467c2).

File: test/test.js (hunk 1 above)
Comment (sulthan-ahmed) [reply test/test.js]: I agree all or none.  I prefer to have it as that's how I learnt JS but whatever not important enough to argue for or against tbh 🤷‍♂

**Comment by zuzak** [comment]
It's `eslint` under the hood, but it's more opinionated -- it checks more rules, I think.
The selling point for me is that it settles all those arbitrary arguments about semicolons or whatever -- you just follow the linter and do what it says.

(And because the codebase is already much closer to standard style than eslint style)

**Comment by sulthan-ahmed** [comment]
At the moment we use eslint for the main app.
Should we be consistent and use standard across all of our repos in that case?

**Comment by zuzak** [comment]
It's not worth the effort to change it on a large existing codebase, but would be good to add it to new projects moving forward.

**Comment by sulthan-ahmed** [comment]
ok

**Pull request merged**